slides/advanced-01-feature-engineering.qmd

```diff
@@ -0,0 +1,783 @@
+```
+
+. . .
+
+To deal with highly correlated predictors, find the minimum set of predictor columns that make the pairwise correlations less than the threshold.
+
+## Other possible steps `r hexes("recipes")`
+
+```{r}
+#| label: pca
+#| code-line-numbers: "6"
+#| eval: false
+hotel_rec <- 
+  recipe(avg_price_per_room ~ ., data = hotel_tr) %>% 
+  step_dummy(all_nominal_predictors()) %>% 
+  step_zv(all_predictors()) %>% 
+  step_normalize(all_numeric_predictors()) %>% 
+  step_pca(all_numeric_predictors())
+```
+
+. . . 
+
+PCA feature extraction...
+
+## Other possible steps `r hexes("recipes", "embed")`
+
+```{r}
+#| label: umap
+#| code-line-numbers: "6"
+#| eval: false
+hotel_rec <- 
+  recipe(avg_price_per_room ~ ., data = hotel_tr) %>% 
+  step_dummy(all_nominal_predictors()) %>% 
+  step_zv(all_predictors()) %>% 
+  step_normalize(all_numeric_predictors()) %>% 
+  embed::step_umap(all_numeric_predictors(), outcome = avg_price_per_room)
+```
+
+. . . 
+
+A fancy machine learning supervised dimension reduction technique...
+
+:::notes
+Note that this uses the outcome, and it is from an extension package
+:::
+
+
+## Other possible steps `r hexes("recipes")`
+
+```{r}
+#| label: splines
+#| eval: false
+#| code-line-numbers: "6"
+hotel_rec <- 
+  recipe(avg_price_per_room ~ ., data = hotel_tr) %>% 
+  step_dummy(all_nominal_predictors()) %>% 
+  step_zv(all_predictors()) %>% 
+  step_normalize(all_numeric_predictors()) %>% 
+  step_spline_natural(year_day, deg_free = 10)
+```
+
+. . . 
+
+Nonlinear transforms like natural splines, and so on!
+
+##  {background-iframe="https://recipes.tidymodels.org/reference/index.html"}
+
+::: footer
+:::
+
+
+## Your turn {transition="slide-in"}
+
+![](images/parsnip-flagger.jpg){.absolute top="0" right="0" width="150" height="150"}
+
+*Create a `recipe()` for the hotel data to :*
+
+-   *use a Yeo-Johnson (YJ) transformation on `lead_time`*
+-   *convert factors to indicator variables*
+-   *remove zero-variance variables*
+
+```{r}
+#| label: your-turn-make-recipe
+#| echo: false
+countdown(minutes = 3, id = "make-recipe")
+```
+
+
+## Minimal recipe `r hexes("recipes")` 
+
+```{r}
+#| label: hotel-rec
+hotel_indicators <-
+  recipe(avg_price_per_room ~ ., data = hotel_tr) %>% 
+  step_YeoJohnson(lead_time) %>%
+  step_dummy(all_nominal_predictors()) %>%
+  step_zv(all_predictors())
+```
+
+## Using a workflow `r hexes("recipes", "workflows", "parsnip", "tune")` {.annotation}
+
+```{r}
+#| label: lm-model
+#| cache: true
+
+set.seed(9)
+
+hotel_lm_wflow <-
+  workflow() %>%
+  add_recipe(hotel_indicators) %>%
+  add_model(linear_reg())
+ 
+ctrl <- control_resamples(save_pred = TRUE)
+hotel_lm_res <-
+  hotel_lm_wflow %>%
+  fit_resamples(hotel_rs, control = ctrl)
+
+collect_metrics(hotel_lm_res)
+```
+
+## Your turn {transition="slide-in"}
+
+![](images/parsnip-flagger.jpg){.absolute top="0" right="0" width="150" height="150"}
+
+*Use `fit_resamples()` to fit your workflow with a recipe.*
```

Comment:
The learners will not necessarily have made their own workflow at this point. Are they supposed to use the one from the previous slides or pick a (different) model to make their own workflow?

Reply:
Maybe I don't understand; we did [make one on the previous slide](https://github.com/tidymodels/workshops/pull/127/files#diff-757f60aaa2bcbca2701235fb4c5e1184fc52674ac8a7d67ea7e79fd5838fcd4bR462)